src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        Node(Node<E> prev, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(null, value, null);
+        } else {
+            if (tail == null) {
+                tail = new Node<>(head, value, null);
+                head.next = tail;
+            } else {
+                tail.next = new Node<>(tail, value, null);
+                tail = tail.next;
+            }
+        }
+        size++;
```

Comment:
If head is null then tail is null too, when size 0 and u add node, u should init head and tail by this node